test/gtest/core/test_multiproc.cc

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+#include "test_multiproc.h"
+extern "C" {
+#include "core/ucc_team.h"
+}
+void test_process::init_lib()
+{
+    EXPECT_EQ(UCC_OK, ucc_lib_config_read(NULL, NULL, &lib_config));
+    lib_params.mask        = UCC_LIB_PARAM_FIELD_THREAD_MODE;
+    lib_params.thread_mode = UCC_THREAD_SINGLE;
+    EXPECT_EQ(UCC_OK, ucc_init(&lib_params, lib_config, &lib_h));
+    ucc_lib_config_release(lib_config);
+}
+
+void test_process::init_ctx()
+{
+    EXPECT_EQ(UCC_OK, ucc_context_config_read(lib_h, NULL, &ctx_config));
+    ctx_params.mask     = UCC_CONTEXT_PARAM_FIELD_TYPE;
+    ctx_params.ctx_type = UCC_CONTEXT_EXCLUSIVE;
+    EXPECT_EQ(UCC_OK,
+              ucc_context_create(lib_h, &ctx_params, ctx_config, &ctx_h));
```

Comment:
Can we move resource creation to common code? As we add more tests, we don’t have to repeat (for example here context creation) the resource creation for each new interface. This might help gtest being relatively faster in the future.

Reply:
It kind of IS in the common code right now. There is a "test_process" class that has the corresponding members. Whenever one wants to add another test with "multiproc" simulation env it can derive from it.